lib/resources/git_resource.py

```diff
@@ -2,19 +2,31 @@
 This module handles Git resources to generate hashs.
 """
 #standard imports
-import glob
 import hashlib
 import os
+import subprocess
 import tempfile
+from stat import S_ISDIR
+from typing import List, Tuple
 
 # third party imports
-from git import Git, Repo, GitCommandError
+from git import GitCommandError, Repo
+from git.diff import Diff, DiffIndex
+from git.objects.commit import Commit
+from git.refs.tag import Tag
 from loguru import logger
 
 # project imports
 from sql.db_connector import DbConnector
 
-EXTENSION = ["txt", "css", "js"]
+FilePath = str
+BlobHash = str
+TagName = str
+FileHash = str
+GitFileMetadata = Tuple[FilePath, TagName, BlobHash]
+FileMetadata = Tuple[FilePath, TagName, FileHash]
+
+BLOB_MODE = 33188
```

Comment:
What is the meaning if this value ? Can you add a comment here or use a more explicit name ? :)